### PR TITLE
[Test] Add unit tests for JetNemotronConfig

### PR DESCRIPTION
**Pull request by MohamedElashri**
## Motivation

Improve CPU-only unit-test coverage for JetNemotronConfig as part of #20865.

## Modifications

Added `test/registered/unit/configs/test_jet_nemotron.py` with coverage for:

- attention and linear layer classification
- empty and mixed layer layouts
- Mamba2CacheParams construction
- TP=1 and TP=2 cache-shape behavior
- non-default expand_v
- Mamba state dtype propagation

And the test itself is registered with `register_cpu_ci()` 

## Accuracy Tests

Not applicable. This is a test-only change.

Local unit-test is done as:

```bash
$ python -m pytest test/registered/unit/configs/test_jet_nemotron.py -v
============================= test session starts ==============================
  platform linux -- Python 3.12.11, pytest-9.1.1, pluggy-1.6.0 -- /data/home/melashri/LLM/sglang/upstream-sglang/.venv/bin/python
  cachedir: .pytest_cache
  rootdir: /data/home/melashri/LLM/sglang/upstream-sglang/test
  configfile: pytest.ini
  plugins: asyncio-1.4.0, anyio-4.14.2, cov-7.1.0
  asyncio: mode=Mode.AUTO, debug=False, asyncio_default_fixture_loop_scope=None, asyncio_default_test_loop_scope=function
  collecting ... collected 6 items

  test/registered/unit/configs/test_jet_nemotron.py::TestJetNemotronConfig::test_expand_v_controls_value_head_dimension PASSED [ 16%]
  test/registered/unit/configs/test_jet_nemotron.py::TestJetNemotronConfig::test_layer_ids_are_empty_for_empty_layout PASSED [ 33%]
  test/registered/unit/configs/test_jet_nemotron.py::TestJetNemotronConfig::test_layer_ids_classify_mixed_layer_types PASSED [ 50%]
  test/registered/unit/configs/test_jet_nemotron.py::TestJetNemotronConfig::test_mamba2_cache_params_preserve_resolved_dtype PASSED [ 66%]
  test/registered/unit/configs/test_jet_nemotron.py::TestJetNemotronConfig::test_mamba2_cache_params_single_tp PASSED [ 83%]
  test/registered/unit/configs/test_jet_nemotron.py::TestJetNemotronConfig::test_mamba2_cache_params_two_way_tp PASSED [100%]

  =============================== warnings summary ===============================
  .venv/lib/python3.12/site-packages/torch/cuda/__init__.py:188
    /data/home/melashri/LLM/sglang/upstream-sglang/.venv/lib/python3.12/site-packages/torch/cuda/__init__.py:188: UserWarning: CUDA initialization: The NVIDIA driver on your system is too
    old (found version 12080). Please update your GPU driver by downloading and installing a new version from the URL: http://www.nvidia.com/Download/index.aspx Alternatively, go to:
    https://pytorch.org to install a PyTorch version that has been compiled with your version of the CUDA driver. (Triggered internally at
    /__w/pytorch/pytorch/c10/cuda/CUDAFunctions.cpp:119.)
      return torch._C._cuda_getDeviceCount() > 0

  python/sglang/kernels/ops/attention/fla/utils.py:223
    /data/home/melashri/LLM/sglang/upstream-sglang/python/sglang/kernels/ops/attention/fla/utils.py:223: UserWarning: Triton is not supported on current platform, roll back to CPU.
      warnings.warn(

  .venv/lib/python3.12/site-packages/torch/jit/_script.py:365: 14 warnings
    /data/home/melashri/LLM/sglang/upstream-sglang/.venv/lib/python3.12/site-packages/torch/jit/_script.py:365: DeprecationWarning: `torch.jit.script_method` is deprecated. Please switch to
    `torch.compile` or `torch.export`.
      warnings.warn(

  python/sglang/srt/layers/quantization/awq/awq.py:52
    /data/home/melashri/LLM/sglang/upstream-sglang/python/sglang/srt/layers/quantization/awq/awq.py:52: UserWarning: Only CUDA, HIP and XPU support AWQ currently.
      warnings.warn(f"Only CUDA, HIP and XPU support AWQ currently.")

  python/sglang/srt/layers/quantization/gguf.py:69
    /data/home/melashri/LLM/sglang/upstream-sglang/python/sglang/srt/layers/quantization/gguf.py:69: UserWarning: Only CUDA, MUSA and NPU support GGUF quantization currently.
      warnings.warn(f"Only CUDA, MUSA and NPU support GGUF quantization currently.")

  -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
  ======================== 6 passed, 18 warnings in 6.76s ========================
```

## Speed Tests and Profiling

Not applicable

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :x: [Run #30675753674](https://github.com/sgl-project/sglang/actions/runs/30675753674)
Latest PR Test (Extra): :x: [Run #30675753506](https://github.com/sgl-project/sglang/actions/runs/30675753506)

### CODE REVIEW

**Comment by hnyls2002** [comment]
Closing this along with the other open unit-test PRs referenced from #20865.

In the AI era, adding unit tests for coverage's sake no longer provides enough value to justify the cost — each PR in this batch consumes reviewer bandwidth and CI time out of proportion to the coverage it adds. The tracking issue itself is already closed, and we are no longer accepting coverage-driven UT PRs from it.

Thanks for the effort — test contributions tied to a concrete bug fix or behavior change are still welcome.
